fulp_modules/features/toys/code/shrimp_plush.dm

```diff
@@ -0,0 +1,37 @@
+/datum/mood_event/shrimp_rice
+	description = "I can't believe a shrimp fried this rice!"
+	mood_change = 3
+	timeout = 2 MINUTES
+
+/obj/item/toy/plush/shrimp/Destroy()
+	. = ..()
+	if(fried_rice)
+		UnregisterSignal(fried_rice, COMSIG_FOOD_EATEN)
+
+/obj/item/toy/plush/shrimp/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	. = ..()
+	if(proximity_flag && has_fried)
+		return
+
+	if(!istype(user.mind.assigned_role, /datum/job/cook))
```

Comment:
This should instead check the title and compare it to the JOB_COOK define, or even better you can check trait for their skillchip.